.github/workflows/api-docs.yml

```diff
@@ -0,0 +1,42 @@
+name: update-api-documentation
+
+on:
+  push:
+    branches: [master]
+    tags: ['v*']
+    paths:
+    - 'docs/**'
+
+jobs:
+  push-to-documentation-branch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Add key
+        env:
+          SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+        run: |
+          mkdir -p ~/.ssh
+          ssh-keyscan github.com >> ~/.ssh/known_hosts
+          echo "${{ secrets.SERVER_DEPLOY_KEY }}" > ~/.ssh/id_rsa
+          chmod 600 ~/.ssh/id_rsa
+          cat <<EOT >> ~/.ssh/config
+          Host github.com
+          HostName github.com
+          IdentityFile ~/.ssh/id_rsa
+          EOT
+      - name: Push generated API documentation to documentation branch
+        env:
+          USE_SSH: true
+          GIT_USER: irrelevant
+        run: |
+          git config --global user.email "irrelevant@completely.irrelevant"
+          git config --global user.name "irrelevant"
+          set -x
```

Comment:
Can you add a comment what this is necessary for?

Reply:
Without preconfiguring git with a username and email, it fails with a non-zero exit code which in turn fails the pipeline. I saw my PR towards facebook/docusaurus merged today with this same approach. A bit verbose but works just fine.

`set -x` configures a container's shell to outputs bash command before executing it. It helps very much to understand the pipeline logs, especially in case smth fails.

It's not a `copy/paste` coding just in case. Or do you mean adding a comment in a pipeline code?

Reply:
As this is not an everyday command (`set -x`) a comment would be helpful for the next person working on it.
Also: Isn't that the default behavior of github actions?

Reply:
Ah, my bad, it's ubiquitously used when you create docker containers and it sticks in my mind since then. I'll add a comment.

I don't think it's the case by default. Check this job: https://github.com/SAP/cloud-sdk/runs/672213308?check_suite_focus=true. The only first one on the list is printed.

Whereas after `set -x` you can see every command output with `+` sign here: https://github.com/SAP/cloud-sdk/runs/672505625?check_suite_focus=true. Helps when debugging and reading job's logs.

Reply:
Cool!